source/heating_model/interface.cc

```diff
@@ -111,45 +132,171 @@ namespace aspect
+                           "is only allowed for reasons of backwards compatibility and "
+                           "can not be used together with the new functionality 'List "
+                           "of model names'. Please remove the 'Include latent heat'"
+                           "setting."));
+        if (include_latent_heat)
+          model_names.push_back("latent heat");
+      }
+      prm.leave_subsection ();
+
+
+
+      // go through the list, create objects and let them parse
+      // their own parameters
+      for (unsigned int name=0; name<model_names.size(); ++name)
+        {
+          heating_model_objects.push_back (std_cxx11::shared_ptr<Interface<dim> >
+                                           (std_cxx11::get<dim>(registered_plugins)
+                                            .create_plugin (model_names[name],
+                                                            "Heating model::Model names")));
+
+          if (SimulatorAccess<dim> *sim = dynamic_cast<SimulatorAccess<dim>*>(&*heating_model_objects.back()))
+            sim->initialize (this->get_simulator());
+
+          heating_model_objects.back()->parse_parameters (prm);
+          heating_model_objects.back()->initialize ();
+        }
     }
 
 
+    template <int dim>
+    void
+    Manager<dim>::update ()
+    {
+      for (typename std::list<std_cxx11::shared_ptr<HeatingModel::Interface<dim> > >::const_iterator
+           heating_model = heating_model_objects.begin();
+           heating_model != heating_model_objects.end(); ++heating_model)
+        {
+          (*heating_model)->update();
+        }
+    }
 
     template <int dim>
-    std::string
-    get_names ()
+    void
+    Manager<dim>::evaluate (const typename aspect::MaterialModel::Interface<dim>::MaterialModelInputs &material_model_inputs,
+                            const typename aspect::MaterialModel::Interface<dim>::MaterialModelOutputs &material_model_outputs,
+                            HeatingModel::HeatingModelOutputs &heating_model_outputs) const
     {
-      return std_cxx11::get<dim>(registered_plugins).get_pattern_of_names ();
+      HeatingModel::HeatingModelOutputs individual_heating_outputs(material_model_inputs.position.size(),
+                                                                   this->n_compositional_fields());
+
+      for (unsigned int q=0; q<heating_model_outputs.heating_source_terms.size(); ++q)
+        {
+          heating_model_outputs.heating_source_terms[q] = 0;
+          heating_model_outputs.lhs_latent_heat_terms[q] = 0;
+        }
+
+      for (typename std::list<std_cxx11::shared_ptr<HeatingModel::Interface<dim> > >::const_iterator
+           heating_model = heating_model_objects.begin();
+           heating_model != heating_model_objects.end(); ++heating_model)
+        {
+          (*heating_model)->evaluate(material_model_inputs, material_model_outputs, individual_heating_outputs);
```

Comment:
it might be worth constructing ```individual_heating_outputs``` so it is filled with default values for each plugin (in case one plugin forgets setting some values)